--- a/modules/xcos/src/java/org/scilab/modules/xcos/palette/PaletteSearchManager.java
+++ b/modules/xcos/src/java/org/scilab/modules/xcos/palette/PaletteSearchManager.java
@@ -32,6 +32,7 @@
 import org.apache.lucene.index.IndexWriterConfig.OpenMode;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.RAMDirectory;
+import org.apache.lucene.util.Version;
 import org.scilab.modules.xcos.palette.model.PaletteBlock;
 import org.scilab.modules.xcos.palette.model.PaletteNode;
 import org.scilab.modules.xcos.palette.model.PreLoaded;
@@ -60,7 +61,7 @@
         try {
             indexedDirectory = new RAMDirectory();
             analyzer = new StandardAnalyzer();
-            IndexWriterConfig config = new IndexWriterConfig(analyzer);
+            IndexWriterConfig config = new IndexWriterConfig(Version.LATEST, analyzer);
             config.setOpenMode(OpenMode.CREATE);
             writer = new IndexWriter(indexedDirectory, config);
         } catch (IOException ex) {

