--- scilab-5.3.3/modules/scicos/src/modelica_compiler/scicosCodeGeneration.ml.orig	2011-08-04 16:13:12.109897627 -0300
+++ scilab-5.3.3/modules/scicos/src/modelica_compiler/scicosCodeGeneration.ml	2011-08-04 16:13:45.393897647 -0300
@@ -1398,7 +1398,7 @@ let generate_code path filename fun_name
     nb_modes
     (List.length model.when_clauses + nb_modes)
     (if model.io_dependency then "true" else "false");
-  Printf.fprintf oc "#include <math.h>\n#include <scicos_block.h>\n";
+  Printf.fprintf oc "#include <math.h>\n#include <scilab/scicos_blocks/scicos_block.h>\n";
   List.iter
     (fun (name, in_types, out_types) ->
       generate_c_function_prototype oc (last name) in_types out_types)
