--- scilab-5.3.3/bin/scilab-adv-cli.orig	2011-08-04 16:04:11.745897858 -0300
+++ scilab-5.3.3/bin/scilab-adv-cli	2011-08-04 16:05:06.537897795 -0300
@@ -667,7 +667,6 @@ check_macosx_version()
 }
 
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -698,6 +697,7 @@ case $OS in
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:/usr/lib/:/usr/lib64/:$LD_LIBRARY_PATH
 
 if test $SCIVERBOSE -ne 0; then
     echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
--- scilab-5.3.3/bin/scilab-cli.orig	2011-08-04 16:05:26.102897839 -0300
+++ scilab-5.3.3/bin/scilab-cli	2011-08-04 16:06:00.196897825 -0300
@@ -667,7 +667,6 @@ check_macosx_version()
 }
 
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -698,6 +697,7 @@ case $OS in
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:/usr/lib/:/usr/lib64/:$LD_LIBRARY_PATH
 
 if test $SCIVERBOSE -ne 0; then
     echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
--- scilab-5.3.3/bin/scilab.orig	2011-08-04 16:06:07.591897812 -0300
+++ scilab-5.3.3/bin/scilab	2011-08-04 16:06:31.691897776 -0300
@@ -667,7 +667,6 @@ check_macosx_version()
 }
 
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -708,6 +707,8 @@ fi
 export LD_LIBRARY_PATH
 export PATH
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:/usr/lib/:/usr/lib64/:$LD_LIBRARY_PATH
+
 if test $IS_SCILAB_BINARY -ne 1; then
 # We are not using the Scilab binary, find out where docbook is.
     find_docbook_path
