--- a/modules/xcos/src/java/org/scilab/modules/xcos/palette/PaletteSearcher.java
+++ b/modules/xcos/src/java/org/scilab/modules/xcos/palette/PaletteSearcher.java
@@ -52,7 +52,7 @@
      */
     public List<Document> search(String str) {
         List<Document> found = new ArrayList<>();
-        try (IndexReader reader = DirectoryReader.open(mgr.getDirectory())) {
+        try (IndexReader reader = DirectoryReader.open(mgr.getDirectory(), true, true)) {
             IndexSearcher searcher = new IndexSearcher(reader);

             StandardQueryParser queryParserHelper = new StandardQueryParser();

